### ISSUE TITLE
Accuracy for Testing and Training

### ISSUE THREAD

**Issue by mlviz-dev** [facebookresearch/dlrm#27]
I am trying to better understand the code and how to get accuracy numbers for testing and training. I have few questions. Please help! 
------------------------------------------------------------------------------------------------------------------------------
Training Command:

Terminal_linux> python dlrm_s_pytorch.py --arch-sparse-feature-size=16 --arch-mlp-bot="13-512-256-64-16" --arch-mlp-top="512-256-1" --data-generation=dataset --data-set=kaggle --processed-data-file=./kaggle_data/kaggleAdDisplayChallenge_processed.npz --loss-function=bce --round-targets=True --learning-rate=0.1 --mini-batch-size=128 --print-freq=1000 --use-gpu --test-freq=1000 --print-time --save-model=model.pt --num-batches=2000
--plot-compute-graph $dlrm_extra_option 2>&1 | tee run_kaggle_pt.log

--num-batches=2000
run pytorch ...
Using 1 GPU(s)...
Loading kaggle dataset...
Reading from pre-processed data=./kaggle_data/kaggleAdDisplayChallenge_processed.npz
Training data
Limiting to 2000 batches of the total  306968 batches
Reading in batch: 2000 / 2000

Testing data
Limiting to 2000 batches of the total  25580 batches
Reading in batch: 2000 / 2000

time/loss/accuracy (if enabled):
Time, Loss, and Accuracy
Finished training it 1000/2000 of epoch 0, 115.30 ms/it, loss 0.517118, **accuracy 0.000 %**
Saving model to model.pt
Testing at - 1000/2000 of epoch 0, loss 0.516816, accuracy 75.354 %, best 75.354 %
Time, Loss, and Accuracy
Finished training it 2000/2000 of epoch 0, 19.67 ms/it, loss 0.509206, **accuracy 0.000 %**
Saving model to model.pt
Testing at - 2000/2000 of epoch 0, loss 0.505281, accuracy 76.267 %, best 76.267 %
done

**Question** - The command I gave - is it training or testing? 
**Question** - Why is it showing 0.0% accuracy for training?
**Question** - How does the script handles training data? How can I run tests for the 7th day Kaggle data that is split between validation and test data?
------------------------------------------------------------------------------------------------------------------------------
Inference command:

python dlrm_s_pytorch.py --arch-sparse-feature-size=16 --arch-mlp-bot="13-512-256-64-16" --arch-mlp-top="512-256-1" --data-generation=dataset --data-set=kaggle --processed-data-file=./kaggle_data/kaggleAdDisplayChallenge_processed.npz --loss-function=bce --round-targets=True --learning-rate=0.1 --mini-batch-size=128 --print-freq=1000 --use-gpu --test-freq=1000 --print-time --load-model=model.pt --inference-only --plot-compute-graph $dlrm_extra_option 2>&1 | tee run_kaggle_pt.log

Loading saved mode model.pt
Saved model Training state: epoch = 0/1, batch = 2000/2000, train loss = 0.509206, train accuracy = 0.000 %
Saved model Testing state: nbatches = 2000, test loss = 0.505281, test accuracy = 76.267 %
time/loss/accuracy (if enabled):
Time, Loss, and Accuracy
Finished inference it 2000/2000 of epoch 0, 50.94 ms/it, loss 0.505281, accuracy 0.000 %

**Question For inference, I didn't get any accuracy. Why is that?
Question How can I get the accuracy for testing and training just like the authors did (picture attached below)?** 

![kaggle_dac_loss_accuracy_plots](https://user-images.githubusercontent.com/54375364/63812568-a13e7d00-c8df-11e9-9aa7-d2dfc6008c81.png)

Thanks!

**Comment by mnaumovfb** [comment]
Let me answer your questions one-by-one:
1. The command you have provided will perform training. However, during the training you can periodically check the accuracy of your model on the test set. How often you check this accuracy is governed by the --test-freq argument. That's why you periodically see test prints during training. This allows you to generate the plots above.
2. It's not clear to me why exactly is your training accuracy showing up as 0.0%. I can not reproduce this on my machine. Try removing some of the arguments from your command line, such as "--plot-compute-graph $dlrm_extra_option", "--save-model=model.pt" and "--use-gpu" to see if this error persists.
3. The data is split into 7 days, the first six are used for training, while the last is used for testing. You can find more details in the data_utils.py and dlrm_data_*.py files about it.
4. I think that accuracy number being 0.0% is related to your earlier problem (see question 2).
5. This has been answered (see question 1).

**Comment by mnaumovfb** [comment]
I will close this issue. Please reopen if you are still running into trouble.
